src/WebJobs.Extensions.SendGrid/Bindings/SendGridHelpers.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Mail;
+using Microsoft.Azure.WebJobs.Extensions.SendGrid;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SendGrid.Helpers.Mail;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Bindings
+{
+    internal class SendGridHelpers
+    {
+        internal static EmailAddress Apply(EmailAddress current, string value)
+        {
+            EmailAddress mail;
+            if (TryParseAddress(value, out mail))
+            {
+                return mail;
+            }
+            return current;
+        }
+
+        internal static bool TryParseAddress(string value, out EmailAddress email)
+        {
+            email = null;
+
+            if (string.IsNullOrEmpty(value))
+            {
+                return false;
+            }
+
+            try
+            {
+                // MailAddress will auto-parse the name from a string like "testuser@test.com <Test User>"
+                MailAddress mailAddress = new MailAddress(value);
+                string displayName = string.IsNullOrEmpty(mailAddress.DisplayName) ? null : mailAddress.DisplayName;
+                email = new EmailAddress(mailAddress.Address, displayName);
+                return true;
+            }
+            catch (FormatException)
+            {
+                return false;
+            }
+        }
+
+        internal static void DefaultMessageProperties(SendGridMessage mail, SendGridConfiguration config, SendGridAttribute attribute)
+        {
+            // Apply message defaulting
+            if (mail.From == null)
+            {
+                if (!string.IsNullOrEmpty(attribute.From))
+                {
+                    EmailAddress from = null;
+                    if (!TryParseAddress(attribute.From, out from))
```

Comment:
nit `out EmailAddress from`

Reply:
Why do you think that helps?

Reply:
it doesn't, especially in the context of how TryParseAddress is used (you could flip around the if statement to be cleaner but eh)
carry on